includes/classes/Authentication/Passwords.php

```diff
@@ -374,4 +384,44 @@ public function enforce_for_user( $user_id ) {
 
 		return $enforce;
 	}
+
+	/**
+	 * Check if password is secure by querying the Have I Been Pwned API.
+	 *
+	 * @param string $password Password to validate.
+	 *
+	 * @return bool True if password is ok, false if it shows up in a breach.
+	 */
+	protected function is_password_secure( $password ): bool {
+		$hash   = strtoupper( sha1( $password ) );
+		$prefix = substr( $hash, 0, 5 );
+		$suffix = substr( $hash, 5 );
+
+		$response = wp_remote_get( self::API_URL . $prefix );
```

Comment:
**issue:** As per https://haveibeenpwned.com/API/v3#UserAgent, we need to specify the User Agent. The docs state:

> The user agent should accurately describe the nature of the API consumer such that it can be clearly identified in the request. Not doing so may result in the request being blocked.

I think specifying `10up Experience WordPress Plugin` should be sufficient, thoughts?

Reply:
Additionally, should we cache this request/response to stop us spamming the API if someone resubmits the same pass multiple times?